<launch>
  <!-- Configure general settings -->
  <env name="ROSCONSOLE_CONFIG_FILE" value="$(find training_coordinator)/config/ros/logging.conf"/>
  <env name="ROSCONSOLE_FORMAT" value="[ ${severity} ${time:%Y-%m-%d %H:%M:%S} ]: ${message}" />
  <param name="use_sim_time" value="False"/>

  <!-- Provide Flatland simulation environment (2D, without agents) -->
  <include file="$(find training_coordinator)/launch/environments/flatland.launch">
    <arg name="show_visualization" value="true"/>
    <arg name="world" default="map_empty"/>
  </include>

  <!-- Training coordinator -->
  <node name="training_coordinator" pkg="training_coordinator" type="test.py" output="screen">
    <param name="pedestrian_model_path" value="$(find training_coordinator)/config/environment/models/pedestrian.yaml" />
    <param name="max_steps_per_episode" value="200" />
    <param name="log_container_path" value="$(find training_coordinator)/logs" />
    <param name="learning_rate" value="3e-3" />
    <param name="total_timesteps" value="100000" />
    <param name="checkpoint_frequency_in_steps" value="10000" />
    <param name="number_logs_to_keep" value="5" />
  </node>

</launch>
